docker/Dockerfile

```diff
@@ -0,0 +1,79 @@
+FROM ubuntu:22.04
+
+ARG TARANTOOL_VERSION
+ARG TARANTOOL_VERSION_REGEX="^([0-9]+\.[0-9]+\.[0-9]+)($|-(alpha|beta|rc)[1-9]$)"
+ARG TARANTOOL_DOWNLOAD_BASE_URL="https://download.tarantool.org/tarantool"
+ARG TARANTOOL_APT_SOURCES_FILE="/etc/apt/sources.list.d/tarantool.list"
+
+ENV TT_APP_NAME="default"
+ENV TT_INSTANCE_NAME="master"
+
+RUN if [ -z "${TARANTOOL_VERSION}" ]; then \
+        echo "ERROR: TARANTOOL_VERSION is not defined"; \
+        echo "ERROR: Please provide, for example, --build-arg TARANTOOL_VERSION=3.1.0"; \
+        exit 1; \
+    elif [ ! $(echo "${TARANTOOL_VERSION}" | grep -E "${TARANTOOL_VERSION_REGEX}") ]; then \
+        echo "ERROR: TARANTOOL_VERSION is not valid"; \
+        echo "ERROR: Please provide, for example, 3.1.0 or 3.1.0-(alpha|beta|rc)(1-9)"; \
+        exit 1; \
+    else \
+        echo "TARANTOOL_VERSION is defined and valid"; \
+    fi
+
+RUN useradd --system --user-group --shell /bin/sh tarantool
+
+RUN apt-get update && \
+    apt-get -y --no-install-recommends install \
+        gosu \
+        curl \
+        gnupg2 \
+        ca-certificates && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN case "${TARANTOOL_VERSION}" in \
+        *-alpha*|*-beta*|*-rc*) \
+            REPO_TYPE="pre-release" \
+            TARANTOOL_VERSION="$(echo "${TARANTOOL_VERSION}" | sed 's/-/~/')" \
+            ;; \
+        *) \
+            REPO_TYPE="release" \
+            ;; \
+    esac && \
+    TARANTOOL_SERIES="series-$(echo "${TARANTOOL_VERSION}" | cut -d . -f 1)" && \
+    UBUNTU_CODENAME="$(. /etc/os-release && echo "${UBUNTU_CODENAME}")" && \
+    curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/gpgkey | apt-key add - && \
+    curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/modules/gpgkey | apt-key add - && \
+    echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/linux-deb/ static main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
+    echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/linux-deb/ static main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
+    echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/ubuntu/ ${UBUNTU_CODENAME} main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
+    apt-get update && \
+    apt-cache show tarantool="${TARANTOOL_VERSION}-1" tarantool-dev="${TARANTOOL_VERSION}-1" && \
+    apt-get -y --no-install-recommends install \
+        tarantool="${TARANTOOL_VERSION}-1" \
+        tarantool-dev="${TARANTOOL_VERSION}-1" \
```

Comment:
Is it affected by the problem described in https://github.com/tarantool/setup-tarantool/commit/4ac2f95de4dfcfc9268e255caabd9287678d13aa? Or the `apt-cache show <...>` command above fails if there is no given version?

Reply:
> Or the apt-cache show <...> command above fails if there is no given version?

Yep.

Proof:
```console
6.986 E: Version '3.0.1-2' for 'tarantool' was not found
6.986 E: Version '3.0.1-2' for 'tarantool-dev' was not found
------
Dockerfile:33
--------------------
  32 |
  33 | >>> RUN case "${TARANTOOL_VERSION}" in \
  34 | >>>         *-alpha*|*-beta*|*-rc*) \
  35 | >>>             REPO_TYPE="pre-release" \
  36 | >>>             TARANTOOL_VERSION="$(echo "${TARANTOOL_VERSION}" | sed 's/-/~/')" \
  37 | >>>             ;; \
  38 | >>>         *) \
  39 | >>>             REPO_TYPE="release" \
  40 | >>>             ;; \
  41 | >>>     esac && \
  42 | >>>     TARANTOOL_SERIES="series-$(echo "${TARANTOOL_VERSION}" | cut -d . -f 1)" && \
  43 | >>>     UBUNTU_CODENAME="$(. /etc/os-release && echo "${UBUNTU_CODENAME}")" && \
  44 | >>>     curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/gpgkey | apt-key add - && \
  45 | >>>     curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/modules/gpgkey | apt-key add - && \
  46 | >>>     echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/linux-deb/ static main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
  47 | >>>     echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/linux-deb/ static main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
  48 | >>>     echo "deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/ubuntu/ ${UBUNTU_CODENAME} main" >> ${TARANTOOL_APT_SOURCES_FILE} && \
  49 | >>>     apt-get update && \
  50 | >>>     apt-cache show tarantool="${TARANTOOL_VERSION}-1" tarantool-dev="${TARANTOOL_VERSION}-1" && \
  51 | >>>     apt-get -y --no-install-recommends install \
  52 | >>>         tarantool="${TARANTOOL_VERSION}-2" \
  53 | >>>         tarantool-dev="${TARANTOOL_VERSION}-2" \
  54 | >>>         tt && \
  55 | >>>     apt-get clean && rm -rf /var/lib/apt/lists/*
  56 |
--------------------
ERROR: failed to solve: process "/bin/sh -c case \"${TARANTOOL_VERSION}\" in         *-alpha*|*-beta*|*-rc*)             REPO_TYPE=\"pre-release\"             TARANTOOL_VERSION=\"$(echo \"${TARANTOOL_VERSION}\" | sed 's/-/~/')\"             ;;         *)             REPO_TYPE=\"release\"             ;;     esac &&     TARANTOOL_SERIES=\"series-$(echo \"${TARANTOOL_VERSION}\" | cut -d . -f 1)\" &&     UBUNTU_CODENAME=\"$(. /etc/os-release && echo \"${UBUNTU_CODENAME}\")\" &&     curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/gpgkey | apt-key add - &&     curl -L ${TARANTOOL_DOWNLOAD_BASE_URL}/modules/gpgkey | apt-key add - &&     echo \"deb ${TARANTOOL_DOWNLOAD_BASE_URL}/${REPO_TYPE}/${TARANTOOL_SERIES}/linux-deb/ static main\" >> ${TARANTOOL_APT_SOURCES_FILE} &&     echo \"deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/linux-deb/ static main\" >> ${TARANTOOL_APT_SOURCES_FILE} &&     echo \"deb ${TARANTOOL_DOWNLOAD_BASE_URL}/release/modules/ubuntu/ ${UBUNTU_CODENAME} main\" >> ${TARANTOOL_APT_SOURCES_FILE} &&     apt-get update &&     apt-cache show tarantool=\"${TARANTOOL_VERSION}-1\" tarantool-dev=\"${TARANTOOL_VERSION}-1\" &&     apt-get -y --no-install-recommends install         tarantool=\"${TARANTOOL_VERSION}-2\"         tarantool-dev=\"${TARANTOOL_VERSION}-2\"         tt &&     apt-get clean && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
```

Reply:
Nice trick!

(I would like it to be commented, you know. Because it is a trick.)

Reply:
OK.